### PR TITLE
Implement update account and change password

### PR DESCRIPTION
**Pull request by hothanhtuan1209**
#21 #22 
In this pull request, I write two API about update information account and change password after login
image
image
image
image

### CODE REVIEW

File: myproject/supermarket_manager/constants.py [hunk 1]
```diff
@@ -52,3 +52,9 @@
 
 # Message for required email or password
 REQUIRED_LOGIN = "Both email and password are required"
+
+# Message for change password
+CHANGE_PASS = "Password changed successfully"
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/constants.py]: Should have been named more clearly, for example: CHANGED_PASSWORD

File: myproject/supermarket_manager/constants.py [hunk 2]
```diff
@@ -52,3 +52,9 @@
...
+
+# Message old password incorrect
+INCORRECT_OLD_PASS = "Old password is incorrect"
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/constants.py]: Similar.

File: myproject/supermarket_manager/views.py [hunk 3]
```diff
@@ -437,3 +441,97 @@ def login_account(request):
...
+
+    Parameters:
+        - request: The HTTP request object.
+        - account_id: The unique identifier of the account whose password will be changed.
+
+    Returns:
+        - JsonResponse: A JSON response indicating the result of the password change operation.
+    """
+
+    if request.method != "POST":
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/views.py]: Should use another method.

**Comment by cuongdoan-agilityio** [comment]
@hothanhtuan1209, Please check your login before making requests.

File: myproject/supermarket_manager/constants.py (hunk 1 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/constants.py]: Done

File: myproject/supermarket_manager/constants.py (hunk 2 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/constants.py]: Done

File: myproject/supermarket_manager/views.py (hunk 3 above)
Comment (hothanhtuan1209) [reply myproject/supermarket_manager/views.py]: Done

File: myproject/supermarket_manager/urls.py [hunk 4]
```diff
@@ -25,4 +27,6 @@
     path("api/accounts", add_account, name="add-account"),
     path("api/accounts/<str:account_id>", get_account_detail, name="get-account-detail"),
     path("api/logins", login_account, name="login-account"),
+    path("api/accounts/updates/<str:account_id>", update_account, name="update-account"),
+    path("api/accounts/<str:account_id>/changes-password", change_password, name="change password")
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/urls.py]: Should update the paths.
I think we no need `/updates/` and `/changes-password`.

Comment (hothanhtuan1209) [reply myproject/supermarket_manager/urls.py]: Done

File: myproject/supermarket_manager/views.py [hunk 5]
```diff
@@ -437,3 +441,97 @@ def login_account(request):
...
+
+    if request.method != "PUT":
+        return JsonResponse({"message": INVALID_METHOD}, status=405)
+    
+    try:
+        account = Account.objects.get(account_id=account_id)
+        data = json.loads(request.body)
+
+        if "status" in data:
+            if account.status == "A":
```
Comment (cuongdoan-agilityio) [inline myproject/supermarket_manager/views.py]: Should create the status enum.

Comment (hothanhtuan1209) [reply myproject/supermarket_manager/views.py]: Done

**Pull request merged**